src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using Azure.Core;
+using Npgsql;
+
+namespace LeanCode.Npgsql.ActiveDirectory;
+
+public static class NpgsqlDataSourceBuilderExtensions
+{
+    public static NpgsqlDataSourceBuilder UseAzureActiveDirectoryAuthentication(
+        this NpgsqlDataSourceBuilder builder,
+        TokenCredential credential
+    )
+    {
+        return builder.UsePeriodicPasswordProvider(
```

Comment:
Can we have a test if this works? :P 

Yeah, yeah, I know, but I think it is high time to have also Postgres-specific tests.

Reply:
What kind of test do you expect? A unit (say non-azure) one with some mocked `TokenCredential` or a E2E one? 
I can think of a test skipped by default, unless you set up credentials with some provisioning script to be run if necessary.

Reply:
If you can develop a unit test, then unit test. If you can't - then integration test that is run on CI will be OK.

Reply:
I did a e2e test connecting to postgres but it requires a databas. We can skip it just like other (send grid/fcm/smsapi) tests and run it only when we touch the integration, or I can set up all the resources in the cloud and run it every time.

Reply:
We should have that on CI (separate workflow) and probably have two targets in the local flow.